content/en/blog/_posts/2023-03-16-kubernetes-1.27-deprecations-and-removals.md

```diff
@@ -0,0 +1,138 @@
+
+The command line flag `enable-taint-manager` for kube-controller-manager is deprecated and will be removed in v1.27. The feature that it supports, taint based eviction, is enabled by default and will continue to be implicitly enabled when the flag is removed. The CLI flag `pod-eviction-timeout` is deprecated and will also be removed together with enable-taint-manager.
+
+### Removal of [CSI Migration](https://github.com/kubernetes/kubernetes/pull/110410) feature gate 
+
+Moving the in-tree volume plugins to out-of-tree CSI drivers. As [CSIMigration](https://github.com/kubernetes/enhancements/issues/625) is GA now. The feature gate will be removed in v1.27.
+
+### Removal of CSIInlineVolume feature gate
+
+The [CSI Ephemeral Volume](https://github.com/kubernetes/kubernetes/pull/111258) feature allows CSI volumes to be specified directly in the pod specification for ephemeral use cases. They can be used to inject arbitrary states, such as configuration, secrets, identity, variables or similar information, directly inside pods using a mounted volume. This feature has graduated to GA in v1.25. Hence the feature gate `CSIInlineVolume` will be removed in this release.
+
+### Removal of EphemeralContainers feature gate
+
+[Emphemeral containers](https://github.com/kubernetes/kubernetes/pull/111402) graduated to GA in v1.25. These are containers with a temporary duration that executes within namespaces of an existing pod. Ephemeral Containers are initiated by a user and intended to observe the state of other pods and containers for troubleshooting and debugging purposes. Hence the `EphemeralContainers` feature gate is always enabled and will be removed from the `--feature-gates` flag on the kube-apiserver and the kubelet command lines.
+
+### Removal of LocalStorageCapacityIsolation  feature gate
+
+The [Local Ephemeral Storage Capacity Isolation](https://github.com/kubernetes/kubernetes/pull/111513) feature moved to GA in v1.25. It provides support for capacity isolation of local ephemeral storage between pods, such as EmptyDir, so that a pod can be hard limited in its consumption of shared resources by evicting Pods if its consumption of local ephemeral storage exceeds that limit. Hence the feature gate `LocalStorageCapacityIsolation` will be removed in this release.
+
+### Removal of [NetworkPolicyEndPort](https://github.com/kubernetes/kubernetes/pull/110868) feature gate
+
+`endPort` in Network Policy promoted to GA in v1.25. Network Policy providers that support `endPort` field that can be used to specify a range of ports to apply a Network Policy. Previously, each Network Policy could only target a single port. So the feature gate `NetworkPolicyEndPort` will be removed in this release.
+
+Please be aware that `endPort` field must be supported by the Network Policy provider. If your provider does not support endPort, and this field is specified in a Network Policy, the Network Policy will be created covering only the port field (single port)
```

Comment:
```suggestion
The v1.25 release of Kubernetes promoted `endPort` in NetworkPolicy to GA. NetworkPolicy providers that support the `endPort` field that can be used to specify a range of ports to apply a NetworkPolicy. Previously, each NetworkPolicy could only target a single port. So the feature gate `NetworkPolicyEndPort` will be removed in this release.
```

Please be aware that `endPort` field must be supported by the Network Policy provider. If your provider does not support endPort, and this field is specified in a Network Policy, the Network Policy will be created covering only the port field (single port)